src/Text/Layout/Table/Cell/WideString.hs

```diff
@@ -0,0 +1,77 @@
+{-# LANGUAGE GeneralisedNewtypeDeriving #-}
+{-# LANGUAGE MultiWayIf                 #-}
+{-# LANGUAGE OverloadedStrings          #-}
+module Text.Layout.Table.Cell.WideString
+    ( WideString(..)
+    , WideText(..)
+    ) where
+
+import Data.String
+import qualified Data.Text as T
+import Text.DocLayout
+
+import Text.Layout.Table.Cell
+import Text.Layout.Table.Primitives.AlignInfo
+import Text.Layout.Table.StringBuilder
+
+-- | A newtype for String in which characters can be wider than one space.
+newtype WideString = WideString String
+    deriving (Eq, Ord, Show, Read, Semigroup, Monoid, IsString, StringBuilder)
+
+instance Cell WideString where
+    dropLeft i (WideString s) = WideString $ dropWide True i s
+    dropRight i (WideString s) = WideString . reverse . dropWide False i $ reverse s
+    visibleLength (WideString s) = realLength s
+    measureAlignment p (WideString s) = measureAlignmentWide p s
+
+    buildCell (WideString s) = buildCell s
+
+dropWide :: Bool -> Int -> String -> String
```

Comment:
Even though it is non-public, please provide at least a sentence of Haskell documentation.